tls-client-pelion/mbedtls_mbedos_pelion_config.h

```diff
@@ -0,0 +1,283 @@
+
+#ifndef MBEDTLS_X509_CSR_WRITE_C
+    #define MBEDTLS_X509_CSR_WRITE_C
+#endif //MBEDTLS_X509_CSR_WRITE_C
+
+#ifndef MBEDTLS_CTR_DRBG_MAX_REQUEST
+    #define MBEDTLS_CTR_DRBG_MAX_REQUEST 2048
+#endif //MBEDTLS_CTR_DRBG_MAX_REQUEST
+
+// Needed by update
+#ifndef MBEDTLS_CIPHER_MODE_CTR
+    #define MBEDTLS_CIPHER_MODE_CTR
+#endif //MBEDTLS_CIPHER_MODE_CTR
+
+// Save ROM and a few bytes of RAM by specifying our own ciphersuite list
+#ifndef MBEDTLS_SSL_CIPHERSUITES
+    #define MBEDTLS_SSL_CIPHERSUITES MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8, \
+                                     MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, \
+                                     MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, \
+                                     MBEDTLS_TLS_PSK_WITH_AES_128_CCM_8, \
+                                     MBEDTLS_TLS_PSK_WITH_AES_256_CCM_8, \
+                                     MBEDTLS_TLS_PSK_WITH_AES_128_CBC_SHA256
+#endif //MBEDTLS_SSL_CIPHERSUITES
+
+/*! All of the following definitions are optimizations (reduce mbedTLS memory usage and size),
+*   changing them is on the user responsibility since they can enlarge
+*   the binary footprint and the memory usage
+*/
+
+// define to save 8KB RAM at the expense of ROM
+#ifndef MBEDTLS_AES_ROM_TABLES
+    #define MBEDTLS_AES_ROM_TABLES
+#endif //MBEDTLS_AES_ROM_TABLES
+
+// Reduce IO buffer to save RAM, default is 16KB
+#ifndef MBEDTLS_SSL_MAX_CONTENT_LEN
+    #define MBEDTLS_SSL_MAX_CONTENT_LEN 4096
+#endif //MBEDTLS_SSL_MAX_CONTENT_LEN
+
+// needed for Base64 encoding Opaque data for
+// registration payload, adds 500 bytes to flash.
+#ifndef MBEDTLS_BASE64_C
+    #define MBEDTLS_BASE64_C
+#endif // MBEDTLS_BASE64_C
+
+// Needed by provisioning
+#undef MBEDTLS_PEM_WRITE_C
+
+// Remove RSA, save 20KB at total
+#if !MBED_CONF_MBED_CLIENT_PAL_RSA_REQUIRED
+    #undef MBEDTLS_RSA_C
+    #undef MBEDTLS_PK_RSA_ALT_SUPPORT
+    #undef MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
+    #undef MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
+    #undef MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED
+#endif // MBED_CONF_MBED_CLIENT_PAL_RSA_REQUIRED
+
+// Remove error messages, save 10KB of ROM
+#undef MBEDTLS_ERROR_C
+
+// Remove selftesting and save 11KB of ROM
+#undef MBEDTLS_SELF_TEST
+
+#undef MBEDTLS_CERTS_C
+
+// Reduces ROM size by 30 kB
+#undef MBEDTLS_ERROR_STRERROR_DUMMY
+
+#undef MBEDTLS_VERSION_FEATURES
+
+#undef MBEDTLS_DEBUG_C
+
+// needed for parsing the certificates
+#undef MBEDTLS_PEM_PARSE_C
+
+#undef MBEDTLS_SHA512_C
+
+#undef MBEDTLS_SSL_SRV_C
+
+#undef MBEDTLS_ECP_DP_SECP192R1_ENABLED
+#undef MBEDTLS_ECP_DP_SECP224R1_ENABLED
+#undef MBEDTLS_ECP_DP_SECP384R1_ENABLED
+#undef MBEDTLS_ECP_DP_SECP521R1_ENABLED
+#undef MBEDTLS_ECP_DP_SECP192K1_ENABLED
+#undef MBEDTLS_ECP_DP_SECP224K1_ENABLED
+#undef MBEDTLS_ECP_DP_SECP256K1_ENABLED
+#undef MBEDTLS_ECP_DP_BP256R1_ENABLED
+#undef MBEDTLS_ECP_DP_BP384R1_ENABLED
+#undef MBEDTLS_ECP_DP_BP512R1_ENABLED
+#undef MBEDTLS_ECP_DP_CURVE25519_ENABLED
```

Comment:
Last time I checked, SHA1 was needed because the website's certificate was using it. I am not sure this has changed, but I would suggest to check and enable it in necessary

Reply:
That's currently being discussed in https://github.com/ARMmbed/mbed-os/issues/8638.

Reply:
It would be good to get that sorted out before merging. Unless we are happy to accept potential CI failures...

Also, I noticed that there were some changes to the certificates, perhaps this is no longer a problem?